src/resources/common.ts

```diff
@@ -1,31 +1,97 @@
+import {
+  Resource as ProtoResource,
+  ResourceType,
+  PolicyResource,
+  ResourceDeclareRequest,
+  ResourceDeclareResponse,
+  ActionMap,
+} from '@nitric/api/proto/resource/v1/resource_pb';
+import resourceClient from './client';
+
+export type Permission = 'reading' | 'writing' | 'deleting';
```

Comment:
Not all resources share these permission types, these will need to be specific to each resource and made generic if we're bringing in a common implementation.